packages/core-contracts/src/contracts/arks/CrossChainArk.sol

```diff
@@ -0,0 +1,403 @@
+            IERC165
+        )
+        returns (bool)
+    {
+        return
+            interfaceId == type(ICrossChainAssetReceiver).interfaceId ||
+            interfaceId == type(ICrossChainStateReadReceiver).interfaceId ||
+            interfaceId == type(IInflightAssetTracking).interfaceId ||
+            interfaceId == type(IERC165).interfaceId;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        INTERNAL FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Boards the Ark by initiating a cross-chain transfer
+     * @param amount Amount of tokens to transfer
+     * @dev This function queues a cross-chain transfer to the target proxy using the registry
+     */
+    function _board(uint256 amount, bytes calldata) internal override {
+        address proxyAddress = _getTargetProxy();
+
+        // Approve BridgeQueue to spend tokens
+        config.asset.approve(address(bridgeQueue), amount);
+
+        bridgeQueue.queueTransferAssets(
+            satelliteChainId,
+            address(config.asset),
+            amount,
+            proxyAddress
+        );
+    }
+
+    /**
+     * @notice Disembarks the Ark by withdrawing assets that are available on the contract
+     * @param amount Amount of tokens to withdraw
+     * @dev This function only validates that enough assets are available on this contract
+     * The actual withdrawal from the satellite chain is processed by a keeper through
+     * FleetProxy.withdrawAndTransfer() which transfers assets back to this contract
+     */
+    function _disembark(uint256 amount, bytes calldata) internal view override {
+        // Ensure we have enough assets on the contract
+        uint256 availableAssets = config.asset.balanceOf(address(this));
+        if (availableAssets < amount) {
+            revert InsufficientAssets(amount, availableAssets);
+        }
+
+        // Note: The actual token transfer is handled by the parent Ark.disembark method
+        // No cross-chain message is required as satellite chain withdrawals are keeper-managed
+    }
+
+    /**
+     * @inheritdoc ICrossChainStateReadReceiver
+     */
+    function receiveStateRead(
+        bytes calldata resultData,
+        address requestor,
+        bytes32 requestId,
+        uint16 sourceChainId
+    ) external {
+        if (msg.sender != address(bridgeRouter)) revert Unauthorized();
+        if (sourceChainId != satelliteChainId) revert InvalidSourceChain();
+        if (requestor != address(this)) revert InvalidRequestor();
+
+        // Decode the remote asset balance
+        uint256 newRemoteBalance = abi.decode(resultData, (uint256));
+
+        lastRemoteAssetBalance = newRemoteBalance;
+
+        emit RemoteAssetBalanceUpdated(lastRemoteAssetBalance, requestId);
+
+        // Reset inflight assets as the state read now reflects the current remote balance
+        inflightAssets = 0;
```

Comment:
looks like it is possible to queuee mutliple bridging operations for a single ark - hence a single state read would erase all `inflightAssets` - for all initiated bridge operations